webapp/src/components/ClassAnalysisTable.tsx

```diff
@@ -0,0 +1,140 @@
+import React from "react";
+import { Box } from "@mui/material";
+import { GridValueFormatterParams, GridCellParams } from "@mui/x-data-grid";
+import { QueryPipelineState } from "types/models";
+import { getClassAnalysisEndpoint } from "services/api";
+import { formatRatioAsPercentageString } from "utils/format";
+import SeeMoreLess, {
+  INITIAL_NUMBER_VISIBLE,
+  useMoreLess,
+} from "components/SeeMoreLess";
+import { Table } from "./Table";
+import { ClassAnalysisClassPair } from "types/api";
+import { isPipelineSelected } from "utils/helpers";
+
+const ROW_HEIGHT = 35;
+const FOOTER_HEIGHT = 40;
+
+type Row = ClassAnalysisClassPair & { id: number };
+type Props = {
+  jobId: string;
+  pipeline: QueryPipelineState;
+};
+const twoDigitFormatter = ({ value }: GridValueFormatterParams) =>
+  isNaN(value as number) ? "--" : (value as number).toFixed(2);
+
+const ClassAnalysisTable: React.FC<Props> = ({ jobId, pipeline }) => {
+  const { data, isFetching, error } = getClassAnalysisEndpoint.useQuery({
+    jobId,
+    ...pipeline,
+  });
+
+  const rows =
+    data?.classPairs.map((classPairs, id) => ({
+      id,
+      ...classPairs,
+    })) || [];
+
+  const { numberVisible, seeMoreLessProps } = useMoreLess({
+    init: INITIAL_NUMBER_VISIBLE,
+    total: rows.length ?? 0,
+  });
+
+  const Footer = () => (
+    <Box
+      height={FOOTER_HEIGHT}
+      display="flex"
+      flexDirection="row"
+      alignItems="end"
+    >
+      <SeeMoreLess {...seeMoreLessProps} />
+    </Box>
+  );
+
+  return (
+    <Table
+      sx={{
+        "& .MuiDataGrid-cell": {
+          borderBottom: "none",
+        },
+        "& .MuiDataGrid-columnHeaders": {
+          borderBottom: "none",
+        },
+        "& .MuiDataGrid-iconSeparator": {
+          display: "none",
+        },
+      }}
+      autoHeight
+      rowHeight={ROW_HEIGHT}
+      columns={[
+        {
+          flex: 1,
+          field: "sourceClass",
+          headerName: "Source Class",
+        },
+        {
+          flex: 1,
+          field: "targetClass",
+          headerName: "Target Class",
+        },
+        {
+          flex: 1,
+          type: "number",
+          field: "overlapScoreTrain",
+          headerName: "Overlap Score",
```

Comment:
This is the table header, right? How do you feel about changing this to "Semantic Overlap Score?" I think "class overlap" is a little confusing in a non-specific, vague way - like, it could be interpreted as something related to model confusion - and "semantic overlap" makes it more clear what we're referring to. I think the title will fit fine on 2 lines, but if it's too long, I think I'd lean towards "Semantic Overlap" over "Overlap Score" (i.e., I think adding _semantic_ is more important than keeping _score_).

```suggestion
          headerName: "Semantic Overlap Score",
```

Reply:
Never mind - we can do this in another PR focusing on improving class analysis docs and in-app communication!